src/main/shortcutHandler.js

```diff
@@ -0,0 +1,245 @@
+      ['paragraphMathBlock', 'CmdOrCtrl+Alt+M'],
+      ['paragraphHtmlBlock', 'CmdOrCtrl+Alt+H'],
+      ['paragraphOrderList', 'CmdOrCtrl+Alt+O'],
+      ['paragraphBulletList', 'CmdOrCtrl+Alt+U'],
+      ['paragraphTaskList', 'CmdOrCtrl+Alt+X'],
+      ['paragraphLooseListItem', 'CmdOrCtrl+Alt+L'],
+      ['paragraphParagraph', 'CmdOrCtrl+0'],
+      ['paragraphHorizontalLine', 'CmdOrCtrl+Alt+-'],
+      ['paragraphYAMLFrontMatter', 'CmdOrCtrl+Alt+Y'],
+
+      // format menu
+      ['formatStrong', 'CmdOrCtrl+B'],
+      ['formatEmphasis', 'CmdOrCtrl+I'],
+      ['formatInlineCode', 'CmdOrCtrl+`'],
+      ['formatStrike', 'CmdOrCtrl+D'],
+      ['formatHyperlink', 'CmdOrCtrl+L'],
+      ['formatImage', 'CmdOrCtrl+Shift+I'],
+      ['formatClearFormat', 'Shift+CmdOrCtrl+R'],
+
+      // window menu
+      ['windowMinimize', 'CmdOrCtrl+M'],
+      ['windowCloseWindow', 'CmdOrCtrl+Shift+W'],
+
+      // view menu
+      ['viewToggleFullScreen', isOsx ? 'Ctrl+Command+F' : 'F11'],
+      ['viewChangeFont', 'CmdOrCtrl+.'],
+      ['viewSourceCodeMode', 'CmdOrCtrl+Alt+S'],
+      ['viewTypewriterMode', 'CmdOrCtrl+Alt+T'],
+      ['viewFocusMode', 'CmdOrCtrl+Shift+F'],
+      ['viewToggleSideBar', 'CmdOrCtrl+J'],
+      ['viewToggleTabBar', 'CmdOrCtrl+Alt+B'],
+      ['viewDevToggleDeveloperTools', isOsx ? 'Alt+Command+I' : 'Ctrl+Shift+I'],
+      ['viewDevReload', 'CmdOrCtrl+R']
+    ])
+
+    // fix non-US keyboards
+    this.mnemonics = new Map()
+    this.fixLayout()
+
+    // load user-defined keybindings
+    this.loadLocalKeybindings()
+  }
+
+  fixLayout () {
+    const lang = getKeyboardLanguage()
+    switch (lang) {
+      // fix aidou and inline code
+      case 'ch':
+      case 'de':
+      case 'dk':
+      case 'fi':
+      case 'no':
+      case 'se':
+        this.fixInlineCode()
+
+        if (!isOsx) {
+          this.fixAidou()
+        }
+        break
+
+      // fix aidou only
+      case 'es':
+      case 'hr':
+      case 'it':
+      case 'pl':
+      case 'pt':
+        if (!isOsx) {
+          this.fixAidou()
+        }
+        break
+
+      // custom layouts
+      case 'bg':
+        this.mnemonics.set('CmdOrCtrl+/', 'CmdOrCtrl+8')
+        this.fixInlineCode()
+        break
+    }
+  }
+
+  fixAidou () {
+    this.mnemonics.set('CmdOrCtrl+/', 'CmdOrCtrl+7')
+  }
+
+  // fix dead backquote key on layouts like German
+  fixInlineCode () {
+    this.keys.set('formatInlineCode', 'CmdOrCtrl+Shift+B')
+  }
+
+  loadLocalKeybindings () {
+    if (global.MARKTEXT_SAFE_MODE || !isFile(this.configPath)) {
+      return
+    }
+
+    const json = readJson(this.configPath, true)
+    if (!json || typeof json !== 'object') {
+      log('Invalid keybindings.json configuration.')
+      return
+    }
+
+    // keybindings.json example:
+    // {
+    //   "fileSave": "CmdOrCtrl+S",
+    //   "fileSaveAs": "CmdOrCtrl+Shift+S"
+    // }
+
+    // TODO: should we check for duplicate shortcuts
```

Comment:
Good idea, we can check the duplicate shortcuts when we add new shortcut.

Reply:
The code is a little messed up.